app/src/main/java/com/graeberj/taskman/core/presentation/navigation/Navigation.kt

```diff
@@ -1,22 +1,39 @@
 package com.graeberj.taskman.core.presentation.navigation
 
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
 import com.graeberj.taskman.auth.presentation.login.LoginScreen
-import com.graeberj.taskman.core.presentation.TaskmanSplashScreen
+import com.graeberj.taskman.auth.presentation.login.LoginViewModel
+import com.graeberj.taskman.auth.presentation.registration.RegistrationScreen
+import com.graeberj.taskman.auth.presentation.registration.RegistrationViewModel
 
 
 @Composable
-fun TaskmanNavigation (window: android.view.Window){
+fun TaskmanNavigation() {
     val navController = rememberNavController()
-    NavHost(navController = navController, startDestination =  Routes.TaskmanSplashScreen.name){
-        composable(Routes.TaskmanSplashScreen.name){
-            TaskmanSplashScreen(navController = navController, window = window)
+    NavHost(navController = navController, startDestination = Routes.LoginScreen.name) {
+
+        composable(Routes.LoginScreen.name) {
+            val viewModel = hiltViewModel<LoginViewModel>()
+            val state by viewModel.state.collectAsState()
+            LoginScreen(
+                state = state,
+                onEvent = viewModel::onEvent,
+                onSignupClick = { navController.navigate(Routes.RegistrationScreen.name) }
+            )
         }
-        composable(Routes.LoginScreen.name){
-            LoginScreen(onSignupClick = {navController.navigate(Routes.RegistrationScreen.name)})
+        composable(Routes.RegistrationScreen.name) {
+            val viewModel = hiltViewModel<RegistrationViewModel>()
+            val state by viewModel.state.collectAsState()
```

Comment:
Recently the collectAsStateWithLifecycle() dependency got stable, so I'd recommend to use that since it's the safest way to collect flows in Android. For that, you'll need this dependency:
```
implementation "androidx.lifecycle:lifecycle-runtime-compose:2.x.x"
```
Then use it like this:
```
val state by viewModel.state.collectAsStateWithLifecycle()
```
If you have a simple flow like here it shouldn't make a difference, but as soon as you convert a normal `Flow` into a `StateFlow` with `stateIn`, the Flow operators will be executed even when the app is in the background (which you typically don't want when dealing with UI updates).